File: truckdoc-client-m2/truckdoc-client-updater/src/main/java/com/sanda/truckdoc/client/updater/network/NetworkModule.java
package com.sanda.truckdoc.client.updater.network;

import com.sanda.truckdoc.client.updater.BuildConfig;

import org.jetbrains.annotations.NotNull;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Created by astra on 07.06.2015.
 */
@Module
public class NetworkModule {

    private static final String TCUS_URI = "http://tcus.truckdoc.ru";

    @Provides
    @Singleton
    @NotNull Backend provideBackend() {
        return new Retrofit.Builder().baseUrl(TCUS_URI)
                .client(new OkHttpClient.Builder().addInterceptor(new HttpLoggingInterceptor().setLevel(BuildConfig.DEBUG ?
                        HttpLoggingInterceptor.Level.BASIC : HttpLoggingInterceptor.Level.NONE))
                        .build())
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJavaCallAdapterFactory.create())
                .build()
                .create(Backend.class);
    }
}

File: app-updater/src/main/java/com/sanda/truckdoc/updater/di/NetworkModule.kt
package com.sanda.truckdoc.updater.di

import android.content.Context
import com.sanda.truckdoc.updater.config.GitHubConfig
import com.sanda.truckdoc.updater.config.JenkinsConfig
import com.sanda.truckdoc.updater.data.api.GitHubApiService
import com.sanda.truckdoc.updater.data.api.JenkinsApiService
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.qualifiers.ApplicationContext
import dagger.hilt.components.SingletonComponent
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import javax.inject.Qualifier
import javax.inject.Singleton

@Qualifier
@Retention(AnnotationRetention.BINARY)
annotation class GitHubRetrofit

@Qualifier
@Retention(AnnotationRetention.BINARY)
annotation class JenkinsRetrofit

@Module
@InstallIn(SingletonComponent::class)
object NetworkModule {
    
    @Provides
    @Singleton
    fun provideOkHttpClient(): OkHttpClient {
        val loggingInterceptor = HttpLoggingInterceptor().apply {
            level = HttpLoggingInterceptor.Level.BODY
        }
        
        return OkHttpClient.Builder()
            .addInterceptor(loggingInterceptor)
            .build()
    }
    
    @Provides
    @Singleton
    @GitHubRetrofit
    fun provideGitHubRetrofit(okHttpClient: OkHttpClient): Retrofit {
        return Retrofit.Builder()
            .baseUrl(GitHubConfig.GITHUB_BASE_URL)
            .client(okHttpClient)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    }
    
    @Provides
    @Singleton
    @JenkinsRetrofit
    fun provideJenkinsRetrofit(okHttpClient: OkHttpClient): Retrofit {
        return Retrofit.Builder()
            .baseUrl(JenkinsConfig.JENKINS_BASE_URL)
            .client(okHttpClient)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    }
    
    @Provides
    @Singleton
    fun provideGitHubApiService(@GitHubRetrofit retrofit: Retrofit): GitHubApiService {
        return retrofit.create(GitHubApiService::class.java)
    }
    
    @Provides
    @Singleton
    fun provideJenkinsApiService(@JenkinsRetrofit retrofit: Retrofit): JenkinsApiService {
        return retrofit.create(JenkinsApiService::class.java)
    }
    
    @Provides
    @Singleton
    fun provideContext(@ApplicationContext context: Context): Context {
        return context
    }
} 